webapp/survey/actions.js

```diff
@@ -84,14 +87,23 @@ export const setActiveSurvey = (surveyId, canEdit = true, dispatchSurveyCreate =
 // ====== UPDATE
 
 export const publishSurvey = () => async (dispatch, getState) => {
-  const surveyId = SurveyState.getSurveyId(getState())
-
-  const { data } = await axios.put(`/api/survey/${surveyId}/publish`)
+  const state = getState()
+  const surveyInfo = SurveyState.getSurveyInfo(state)
+  const lang = AppState.getLang(state)
+  const surveyLabel = Survey.getLabel(surveyInfo, lang)
 
   dispatch(
-    showAppJobMonitor(data.job, async () => {
-      await dispatch(setActiveSurvey(surveyId, true))
-      await dispatch(initSurveyDefs(true, true))
+    showDialogConfirm('homeView.surveyInfo.confirmPublish', { survey: surveyLabel }, async () => {
```

Comment:
would it make more sense to have a ButtonPublishSurvey in common components rather than having the showDialogConfirm here? I'd rather leave the action pure, it just publishes. the confirm is more a UI responsibility. what do u think?

Reply:
you are right, but will the buttons have the same style? Also, will you call the action `publish` without a confirm?

Reply:
why same class? can't you pass className to ButtonSurveyPublish?
The confirm can go inside ButtonSurveyPublish, that's what I meant by ```the confirm is more a UI responsibility```

Reply:
Will they have the same icon too?

Reply:
yes. same icon

Reply:
ok